geps/gep-3875/index.md

```diff
@@ -0,0 +1,192 @@
+# GEP-3875: BackendTLS consumer overrides and context awareness
+
+* Issue: [#3875](https://github.com/kubernetes-sigs/gateway-api/issues/3875)
+* Status: Provisional
+
+
+## TLDR
+
+This GEP aims to make the configuration of `BackendTLSPolicy` more flexible by allowing users to specify when the policy should be applied.
+This extends the current state of the API, where this can only be controlled based on Service but the application developer persona.
+
+## Goals
+
+* Allow [service consumers](https://gateway-api.sigs.k8s.io/concepts/glossary/?h=gloss#consumer-route) to define their own TLS policies (the current state would be described as a [service producer policy](https://gateway-api.sigs.k8s.io/concepts/glossary/?h=gloss#producer-route)).
+* Allow contextual selection of a TLS policy. For instance, allow usage of TLS from a specific Gateway or HTTPRoute but not others.
+
+## Non-Goals
+
+(What is out of scope for this proposal.)
+
+## Introduction
+
+(Can link to external doc -- but we should bias towards copying
+the content into the GEP as online documents are easier to lose
+-- e.g. owner messes up the permissions, accidental deletion)
+
+## API
+
+This proposal includes a few API changes:
+
+1. A new `namespaces` field will be added to `spec.targetRefs`.
+2. A new `from` field will be added to `spec.targetRefs`.
+3. A new `mode` field will be added to `spec`.
+
+### `TargetRef.Namespace`
+
+Implementation target: Experimental in 1.4.
+
+The `namespace` field allows a service consumer in another namespace to override the BackendTLSPolicy to use when connecting to a Service.
+
+* `BackendTLSPolicy` can be applied to a `Service` in the same namespace to provide defaults to clients.
+* `BackendTLSPolicy` can be applied to a `Service` in a different namespace to provide explicit configuration for clients in that namespace.
+
+In a complex setup, there are up to 4 namespaces involved: the Gateway, ListenerSet, Route, and Service.
+However, `BackendTLSPolicy` will only be consulted in the `Gateway` and `Service` namespace.
+For mesh use cases, there are only 2 relevant namespaces: the calling client namespace and the `Service` namespace; both of these will be consulted.
```

Comment:
can you be more specific how `the calling client namespace` should be expressed with Gateway API?